src/core/Track.cpp

```diff
@@ -551,11 +551,13 @@ void Track::removeBar( const TimePos & pos )
  */
 void Track::updateLength()
 {
+	if (!Engine::getSong()) { return; }
```

Comment:
Won't this result in not updating `m_length` at all? Is this acceptable?

Reply:
The song length is only used for determining the length of the horizontal scrollbar, and for determining the end of the song when exporting without loop markers.  So this shouldn't cause any problems whatsoever, unless it's somehow possible to export songs without there being an active Song.  I'm not sure how LMMS works in the command line, but I assume it still creates a Song when doing its magic stuff, right?

Reply:
> I'm not sure how LMMS works in the command line, but I assume it still creates a Song when doing its magic stuff, right?

`main` calls `Engine::init` when exporting a projects from CLI, so yes.

Reply:
>  So this shouldn't cause any problems whatsoever, unless it's somehow possible to export songs without there being an active Song.

True, but it's always good to keep the state of an object consistent.